orttraining/orttraining/python/training/api/provider_type.py

```diff
@@ -0,0 +1,27 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# _provider_type.py
+
+from typing import NewType, Text
+
+
+class DeviceType(object):
+    _Type = NewType("_Type", int)
+    cpu = _Type(0)  # type: _Type
+    cuda = _Type(1)  # type: _Type
```

Comment:
Why do we need this? Can we use enum instead?

Reply:
Using type will handle the validation better, so we don't have to worry about checking if the value is in the enums
this is also being used in onnx : https://github.com/onnx/onnx/blob/main/onnx/backend/base.py